Update to 9.900.3 with macOS BLAS, Azure fixes

henryiii opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
*  Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

This is an alternative to #17  with the upstream patch applied for macOS BLAS detection.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

henryiii commented:
See https://gitlab.com/conradsnicta/armadillo-code/issues/139 for the original discussion and patch.

henryiii commented:
The azure failure is not related to this PR, as appveyor is building on Windows correctly. It does seem like the blas search is still having issues, however.

henryiii commented:
@isuruf, is the problem here that it is specifically looking for OpenBLAS and not BLAS in general? Not very familiar with this yet. It's a custom search module instead of FindBLAS.cmake, it's using [ARMA_FindOpenBLAS.cmake](https://gitlab.com/conradsnicta/armadillo-code/blob/9.800.x/cmake_aux/Modules/ARMA_FindOpenBLAS.cmake) and not [ARMA_FindBLAS.cmake](https://gitlab.com/conradsnicta/armadillo-code/blob/9.800.x/cmake_aux/Modules/ARMA_FindBLAS.cmake), AFAICT.

henryiii commented:
@conda-forge-admin please rerender

henryiii commented:
I think this is now working, besides the broken azure build. Does anyone know why that is/was broken?

henryiii commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I tried to re-render for you, but it looks like there was nothing to do.

henryiii commented:
I've fixed bld.bat to work with `Azure`, and dropped the custom config, it should just build normally in Azure now. @isuruf, should I drop the travis/appveyor secrets? I'm assuming they are not useful anymore.

Should be ready for review and to merge.

conradsnicta commented:
@henryiii - There's no need for the patch. Armadillo 9.800.3 (released yesterday) already ships with the modified CMakeLists.txt

marcelotrevisani commented on recipe/meta.yaml:
```diff
@@ -7,17 +7,17 @@ package:
 
 build:
-  number: 1204
+  number: 0
```
Could you please add a few things on ``requirements`` ``build``?
It would be better to add ``make`` for ``unix`` and ``nmake`` for ``windows``
```yaml
requirements:
  build:
    - make   # [unix]
    - nmake  # [win]
```

marcelotrevisani replied:
As this is a new version, could you please double-check if it still needs to pin ``superlu``?
``superlu 5.2.* ``

marcelotrevisani replied:
I check on their docs and it seems it is still necessary to pin ``superluc 5.2.*``

marcelotrevisani commented:
> @henryiii - There's no need for the patch. Armadillo 9.800.3 (released yesterday) already ships with the modified CMakeLists.txt

Hi @henryiii , could you please bump it then to ``9.800.3`` ?

henryiii commented:
Feedback addressed, builds passing.

henryiii commented:
@conradsnicta, would you be interested in being added to the maintainers of this recipe, so you would be able to make changes if needed? It's always nice to have at least one upstream author in the maintainer list.

grlee77 commented:
Hi @henryiii, thanks for updating this. It would be great to have you as a co-maintiner if you can add yourself to the `meta.yaml`.

One odd thing in looking near the end of the OSX or Linux logs on Azure, I see some warnings that look potentially troubling:
```
The following NEW packages will be INSTALLED:

    armadillo:   9.800.3-hcb88e47_0               local      
    arpack:      3.6.3-blas_openblashc42ff08_1002 conda-forge [blas_openblas]
    blas:        1.1-openblas                     conda-forge
    libblas:     3.8.0-11_openblas                conda-forge
    libcxx:      9.0.0-h89e68fa_1                 conda-forge
    libgfortran: 3.0.1-0                          conda-forge
    liblapack:   3.8.0-11_openblas                conda-forge
    libopenblas: 0.3.6-hdc02c5d_2                 defaults   
    openblas:    0.3.3-hdc02c5d_1001              conda-forge
    superlu:     5.2.1-blas_openblashc42ff08_1203 conda-forge [blas_openblas]

Preparing transaction: ...working... done
Verifying transaction: ...working... 
ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: defaults::libopenblas-0.3.6-hdc02c5d_2, conda-forge::openblas-0.3.3-hdc02c5d_1001, conda-forge::libblas-3.8.0-11_openblas
  path: 'lib/libblas.dylib'


ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: defaults::libopenblas-0.3.6-hdc02c5d_2, conda-forge::openblas-0.3.3-hdc02c5d_1001
  path: 'lib/libcblas.dylib'


ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: defaults::libopenblas-0.3.6-hdc02c5d_2, conda-forge::openblas-0.3.3-hdc02c5d_1001, conda-forge::liblapack-3.8.0-11_openblas
  path: 'lib/liblapack.dylib'


ClobberWarning: This transaction has incompatible packages due to a shared path.
  packages: defaults::libopenblas-0.3.6-hdc02c5d_2, conda-forge::openblas-0.3.3-hdc02c5d_1001
  path: 'lib/libopenblas.dylib'



done
Executing transaction: ...working... 
ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/openblas-0.3.3-hdc02c5d_1001/lib/libblas.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/libblas.dylib




ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/openblas-0.3.3-hdc02c5d_1001/lib/libcblas.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/libcblas.dylib




ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/openblas-0.3.3-hdc02c5d_1001/lib/liblapack.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/liblapack.dylib




ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/openblas-0.3.3-hdc02c5d_1001/lib/libopenblas.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/libopenblas.dylib




ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/libblas-3.8.0-11_openblas/lib/libblas.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/libblas.dylib




ClobberWarning: Conda was asked to clobber an existing path.
  source path: /usr/local/miniconda/pkgs/liblapack-3.8.0-11_openblas/lib/liblapack.dylib
  target path: /usr/local/miniconda/conda-bld/armadillo_1575643834760/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/liblapack.dylib

```


Do you know if this is something to be concerned about? I see that there is one `libopenblas` getting pulled in from defaults for some reason.

henryiii commented:
@grlee77, sure.

I'm not sure about the blas clobbering. I *think* it may be okay since those files in the blas recipe should be the same as the ones in the openblas defaults, but I could be completely wrong. @isuruf helped set this up in March, he might have a suggestion?

henryiii commented:
Both of the versions (`openblas`, `libopenblas`) are a bit older, not sure why it's not picking up 0.3.7 from this one: https://github.com/conda-forge/openblas-feedstock/blob/master/recipe/meta.yaml

isuruf commented:
```
    arpack:      3.6.3-blas_openblashc42ff08_1002 conda-forge [blas_openblas]
    superlu:     5.2.1-blas_openblashc42ff08_1203 conda-forge [blas_openblas]
```
Above two packages are using the old `blas` scheme. Do you know why that is?

grlee77 commented:
> Above two packages are using the old blas scheme. Do you know why that is?

I do not. I am the maintainer for `superlu` so I can try to update that recipe. It looks like @jschueller is the maintainer for `arpack`.

isuruf commented:
Both of them have new build numbers that use the new blas scheme. I just want to know why these old versions were picked up

grlee77 commented:
@isuruf: actually for superlu, it looks like you merged an autotick PR related to updating BLAS (conda-forge/superlu-feedstock#9). perhaps something went wrong there?

grlee77 commented:
Actually the old versions were not used during the build, it is just during the test phase that it is installing the older versions for some reason. 

I tried building the recipe locally on linux, but did not encounter the same problem. It found the latest version or arpack & superlu during both build and test phases.

wolfv commented:
@henryiii should we give this another try?

henryiii commented:
Sure, though it needs a version bump at this point. Will keep this page open and hopefully remember to try in a few hours.

marcelotrevisani commented:
Could you please solve the conflicts?

henryiii commented:
Yes, working on it, but I'm home alone with a 4 year old, so even a 5 minute job takes hours. :)

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict.
Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

henryiii commented:
@conda-forge-admin, please rerender

henryiii commented:
If I'm lucky, it will rerender and that will help... Otherwise, I'll have to set up an azure token it seems.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

henryiii commented:
Ready, I think.

grlee77 commented:
Looks good, thanks @henryiii

wolfv commented:
awesome! thanks!
